libs/libc/misc/lib_utsname.c

```diff
@@ -93,7 +93,8 @@ int uname(FAR struct utsname *name)
 
   strlcpy(name->release,  CONFIG_VERSION_STRING, sizeof(name->release));
 
-#if defined(__DATE__) && defined(__TIME__)
+#if defined(__DATE__) && defined(__TIME__) && \
+    !defined(CONFIG_NSH_DISABLE_UNAME_TIMESTAMP)
```

Comment:
should we move the Konfig to kernel side? it isn't good to depend on apps's Kconfig inside nuttx.

Reply:
BTW, it's better to skip the follow line:
https://github.com/apache/nuttx/blob/master/libs/libc/misc/Make.defs#L79-L82
instead skip the print.

Reply:
You mean:

```
#if !defined(CONFIG_UNAME_DISABLE_TIMESTAMP)
.PHONY: lib_utsname$(OBJEXT)
#endif
```

Reply:
Yes, but it's the config name isn't good, since timestamp still exist.

Reply:
Actually I did some tests here with this approach and I think it is better to follow my initial idea because after "make clean" and now "make" the binaries becomes different again.